src/databases/Blueprint/avtBlueprintFileFormat.C

```diff
@@ -870,6 +928,83 @@ avtBlueprintFileFormat::ReadRootFile()
 
 }
 
+// ****************************************************************************
+//  Method: avtBlueprintFileFormat::ReadRootIndexItems
+//
+//  Purpose: Read only the metadata portions of the root file that VisIt needs.
+//
+//  Programmer: cyrush
+//  Creation:   Mon Mar  9 12:24:16 PDT 2020
+//
+//  Modifications:
+//
+//
+// ****************************************************************************
+void
+avtBlueprintFileFormat::ReadRootIndexItems(const std::string &root_fname,
+                                           const std::string &root_protocol,
+                                           conduit::Node &root_info)
+{
+    // list of names we want to read from the root file
+    conduit:Node index_names;
+    index_names.append() = "blueprint_index";
+    index_names.append() = "file_pattern";
+    index_names.append() = "tree_pattern";
+    index_names.append() = "number_of_trees";
+    index_names.append() = "number_of_files";
+    index_names.append() = "protocol";
+
+    if(root_protocol == "hdf5")
+    {
+        // we can't use relay::io::IOHandle here b/c it currently
+        // lacks an option to open the file in read only mode, and 
+        // by default opens in R/W.
+        //
+        // For the case where the data is also included in the root file,
+        // VisIt may have a read only hdf5 file handle open, trying 
+        // to open with R/W will throw an error.
+        // 
+        // we still want partial I/O, so we use the conduit interface
+        // the uses hdf5 handles directly
+        hid_t h5_id = relay::io::hdf5_open_file_for_read(root_fname);
```

Comment:
Just curious but is/was `relay::io::hdf5_open_file_for_read()` actually opening the HDF5 file with `H5F_ACC_RDWR`?

Reply:
it is not. `relay::io::hdf5_open_file_for_read` does what you would expect. 

But the IOHandle path would, that prevents me from using it until I add an option to open read only .

The helper that was checking if the file was hdf5 by opening it was also using `H5F_ACC_RDWR`, when it doesn't need to.